core/src/main/java/io/lionweb/lioncore/java/model/Node.java

```diff
@@ -117,14 +98,25 @@ default Object getPropertyValueByID(String propertyID) {
 
   /** Return a list containing this node and all its descendants. */
   default @Nonnull List<Node> thisAndAllDescendants() {
-    List<Node> nodes = new ArrayList<>();
-    nodes.add(this);
-    for (Node child : this.getChildren()) {
+    List<Node> result = new ArrayList<>();
+    selfAndAllDescendants(this, false, result);
+    return result;
+  }
+
+  /** Return a list containing this node and all its descendants. */
+  static @Nonnull <T extends ClassifierInstance<?>> void selfAndAllDescendants(
```

Comment:
I am not sure why this is a static method and why it is a name very similar to the instance method above

Reply:
Should this be move to ClassiffierInstance, renamed `thisAndAllDescendants` and be made an instance method?

Reply:
I renamed it because Java had issues discerning the different overloads.
`ClassifierInstance` is probably a better place, yes.